File: leetcode-problems/0001-two-sum/solution.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int l=nums.length;
        for (int i=0;i<l;i++){
            for (int j=0;j<l;j++){
                if (i!=j){
                    if (nums[i]+nums[j]==target){
                        int[] ans={i,j};
                        return ans;
                    }
                }
            }
        }
        return new int[]{};
    }
}

File: leetcode-problems/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> ump;
        for (int i=0;i<nums.size();i++){
            int othere=target-nums[i];
            if (ump.find(othere)!=ump.end()) return {ump[othere],i};

            ump[nums[i]]=i;
        }
        return {};
    }
};
